### ISSUE TITLE
Crash on recon_alloc:memory(used) on OTP 23.0

### ISSUE THREAD

**Issue by mworrell** [ferd/recon#82]
On requesting the used memory on OTP 23.0:

```
(zotonic@PoToi)2> recon_alloc:memory(used).
** exception error: bad argument
     in function  element/2
        called as element(2,false)
     in call from recon_alloc:container_value/4 (/Users/marc/Sites/zotonic-master/_build/default/lib/recon/src/recon_alloc.erl, line 696)
     in call from recon_alloc:container_size/3 (/Users/marc/Sites/zotonic-master/_build/default/lib/recon/src/recon_alloc.erl, line 668)
     in call from lists:map/2 (lists.erl, line 1239)
     in call from recon_alloc:memory/2 (/Users/marc/Sites/zotonic-master/_build/default/lib/recon/src/recon_alloc.erl, line 208)
```

This is on macOS 10.14.6, OTP 23.0, build with kerl:

```
Erlang/OTP 23 [erts-11.0] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [hipe]
```

**Comment by lukebakken** [comment]
For what it's worth, I can't reproduce this on Arch Linux. Erlang 23.0.3 built with `kerl`, using the `master` branch of `recon`:

```
(23.0.3)lbakken@shostakovich ~/development/ferd/recon (master=)
$ rebar3 shell
===> Verifying dependencies...
===> Compiling recon                                                                                                                                                                                                                                                            
Erlang/OTP 23 [erts-11.0.3] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe]                                                                                                                                                                                    

Eshell V11.0.3  (abort with ^G)
1> recon_alloc:memory(used).
23599352
2>
```

What version of `recon` are you using?

**Comment by ferd** [comment]
Probably related to https://github.com/ferd/recon/pull/78

call `rebar3 update` before upgrading recon to invalidate the local cache for the app versions stored.

**Comment by mworrell** [comment]
Indeed - fixed somewhere between 2.4.0 and 2.5.1.

Even after upgrade the system somehow kept using 2.4.0, gave it a good kick and now with 2.5.1 it is fixed.

Thanks!
